soda/scientific/soda/scientific/anomaly_detection/anomaly_detector.py

```diff
@@ -0,0 +1,263 @@
+            df,
+            df_flattened,
+            left_index=True,
+            right_index=True,
+            suffixes=("", "_diag"),
+        )
+        return df_joined
+
+    def _parse_params(self) -> Dict[str, Any]:
+        try:
+            this_dir = Path(__file__).parent.resolve()
+
+            # Read detector configuration
+            with open(this_dir.joinpath("detector_config.yaml")) as stream:
+                loaded_config = yaml.safe_load(stream)
+
+            # Manipulate configuration
+            loaded_config["response_params"]["output_columns"] = self._replace_none_values_by_key(
+                loaded_config["response_params"]["output_columns"]
+            )
+            loaded_config["feedback_processor_params"]["output_columns"] = self._replace_none_values_by_key(
+                loaded_config["feedback_processor_params"]["output_columns"]
+            )
+            self._logs.debug(f"Config parsed {loaded_config}")
+
+            return loaded_config
+
+        except Exception as e:
+            self._logs.error(e)
+            raise e
+
+    @staticmethod
+    def _replace_none_values_by_key(dct: Dict[str, Any]) -> Mapping[str, Any]:
+        result = {}
+        for key, value in dct.items():
+            if value is None:
+                value = key
+            result[key] = value
+        return result
+
+    @staticmethod
+    def _parse_output(
+        df_anomalies: pd.DataFrame, freq_detection_result: FreqDetectionResult
+    ) -> Tuple[str, Dict[str, Any]]:
+        if not df_anomalies.empty:
+            results_dict = df_anomalies.to_dict(orient="records")[0]
+            level = results_dict["level"]
+            diagnostics = {
+                "value": results_dict["real_data"],
+                "warn": {
+                    "greaterThanOrEqual": results_dict["warning_greater_than_or_equal"],
+                    "lessThanOrEqual": results_dict["warning_lower_than_or_equal"],
+                },
+                "fail": {
+                    "greaterThanOrEqual": results_dict["critical_greater_than_or_equal"],
+                    "lessThanOrEqual": results_dict["critical_lower_than_or_equal"],
+                },
+                "anomalyProbability": results_dict["anomaly_probability"],
+                "anomalyPredictedValue": results_dict["trend"],
+                "anomalyErrorSeverity": freq_detection_result.error_severity,
+                "anomalyErrorCode": freq_detection_result.error_code,
+            }
+        else:
+            level = "pass"
+            diagnostics = {
+                "value": None,
+                "warn": None,
+                "fail": None,
+                "anomalyProbability": None,
+                "anomalyPredictedValue": None,
+            }
+
+        diagnostics_dict: Dict[str, Any] = AnomalyDiagnostics.parse_obj(diagnostics).dict()
+        return level, diagnostics_dict
+
+    def evaluate(self) -> Tuple[str, Dict[str, Any]]:
```

Comment:
Very minor thing, but to increase the readability, we can move this method just next to __init method (i.e line 115)